src/js/stores/account.js

```diff
@@ -26,29 +27,26 @@ export default Reflux.createStore({
     const usr = encodeURIComponent(data.username)
     const psw = encodeURIComponent(data.password)
     const eml = encodeURIComponent(data.email)
+    const name = encodeURIComponent(data.name)
 
     fetch(`${proxy}/register`, {
       method: 'POST',
-      body: `username=${usr}&password=${psw}&email=${eml}`,
+      body: `username=${usr}&password=${psw}&name=${name}&email=${eml}`,
       headers: {
         'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
       }
-    })
-    .then((res) => {
+    }).then((res) => {
       if (res.status === 400) {
         throw new Error('USERNAME_TAKEN')
       }
-      res.json().then(() => {
+      this.state = {
+        emailVerifyScreen: true
+      }
+      this.trigger(this.state)
+      res.json().then((js) => {
         if (data.name || data.email) {
-          let payload = {
-            name: data.name,
-            email: data.email
-          }
-          Account.login(data.username, data.password, payload)
-        } else {
-          Account.login(data.username, data.password)
+          this.onSetNameEmail(js.webid, data.name, data.email)
```

Comment:
This function does not work anymore, because we attempt to send a patch request to an account that has not been 'activated' yet, therefore the requests fail.